pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -464,21 +464,40 @@ public static SelectionResults renderSelectionResultsWithoutOrdering(List<Object
    *
    * @param rows selection rows.
    * @param dataSchema data schema.
+   * @param selectionColumns selection columns.
    * @return {@link ResultTable} object results.
    */
-  public static ResultTable renderResultTableWithoutOrdering(List<Object[]> rows, DataSchema dataSchema) {
+  public static ResultTable renderResultTableWithoutOrdering(List<Object[]> rows, DataSchema dataSchema, List<String> selectionColumns) {
```

Comment:
Can we add some optimization by checking `selectionColumn` size and `dataSchema`size?

If their size is equal, we don't need to go through the recreation of objects. We will be able to save some memory resources there. Since this happens for each query, it's good to add the optimization if possible.

Reply:
Yes, updated.